packages/starlight/utils/navigation.ts

```diff
@@ -134,8 +141,15 @@ function makeLink(
 	badge?: Badge,
 	attrs?: LinkHTMLAttributes
 ): Link {
-	if (!isAbsolute(href)) href = pathWithBase(href);
-	const isCurrent = href === ensureTrailingSlash(currentPathname);
+	href = formatPath(href, {
+		format: project.build.format,
+	});
```

Comment:
Should we not preserve the previous behaviour and avoid formatting absolute URLs? I would guess we shouldn’t enforce our formatting on an external URL. We’re not currently testing any sidebar entries with `{ link: "https://example.com" }`, so may have missed changes in formatting here.

```suggestion
	if (!isAbsolute(href)) {
		href = formatPath(href, { format: project.build.format });
	}
```